checkov/common/checks_infra/solvers/attribute_solvers/base_attribute_solver.py

```diff
@@ -151,10 +151,13 @@ def _process_node(
         else:
             failed_vertices.append(data)
 
+    def is_needs_all_condition(self) -> bool:
+        return self.is_jsonpath_check
+
     def _evaluate_attribute_matches(
         self, vertex: dict[str, Any], attribute_matches: list[str], filtered_attribute_matches: list[str]
     ) -> bool | None:
-        if self.is_jsonpath_check:
+        if self.is_needs_all_condition():
```

Comment:
shouldn't we always check everything by default? or every time we use a "list" check for example?

Reply:
No we shouldn't. For example for "contains" we should keep the old behavior where we need just one match. That's why I added this exception for "not_contains" only.